File: Python3/add_binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        res = ''
        i = len(a)-1
        j = len(b)-1
        hold = 0
        while i>=0 or j>=0 or hold:
            if i >= 0: hold += (int(a[i]))
            if j >= 0: hold += (int(b[j]))
            res += (str(hold%2))
            hold //= 2
            i -= 1
            j -= 1
        return res[::-1]

File: C++/add_binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res;
        int i = a.size()-1;
        int j = b.size()-1;
        int hold = 0;
        while(i >= 0 || j >= 0 || hold) {
            if (i >= 0) hold += a[i--] - '0';
            if (j >= 0) hold += b[j--] - '0';
            res += hold % 2 + '0';
            hold /= 2;
        }
        reverse(begin(res), end(res));
        return res;
    }
};
